s3torchbenchmarking/src/s3torchbenchmarking/benchmark_utils.py

```diff
@@ -19,16 +25,67 @@
     nvmlInit()
 
 
+class Distribution:
+    def __init__(self, initial_capacity: int, precision: int = 4):
+        self.initial_capacity = initial_capacity
+        self._values = np.zeros(shape=initial_capacity, dtype=np.float32)
+        self._idx = 0
+        self.precision = precision
+
+    def _expand_if_needed(self):
+        if self._idx > self._values.size - 1:
+            self._values = np.concatenate(
+                self._values, np.zeros(self.initial_capacity, dtype=np.float32)
+            )
+
+    def add(self, val: float):
+        self._expand_if_needed()
+        self._values[self._idx] = val
+        self._idx += 1
+
+    def summarize(self) -> dict:
+        window = self._values[: self._idx]
+        if window.size == 0:
+            return {}
```

Comment:
Why the empty dict here?

Reply:
The call to `window.mean()` below would fail when working with an empty slice. In practise, this should never be empty though so maybe it's okay to let it fail? I thought this was a better fallback behavior.

Reply:
It's fine, I was just wondering if `None` or a dict with the expected keys but all set to say 0 or NaN. As-is is good